compiler/rustc_lint/src/builtin.rs

```diff
@@ -2833,7 +2833,7 @@ impl ClashingExternDeclarations {
                 return true;
             }
             let tcx = cx.tcx;
-            if a == b || rustc_middle::ty::TyS::same_type(a, b) {
+            if a == b {
                 // All nominally-same types are structurally same, too.
```

Comment:
nit: "nominally" here is potentially confusing, maybe good to adjust this to say "literally the same type" or similar?